pkg/auth/handler/webapp/authflowv2/reset_password.go

```diff
@@ -75,10 +110,87 @@ func (h *AuthflowV2ResetPasswordHandler) GetErrorData(w http.ResponseWriter, r *
 }
 
 func (h *AuthflowV2ResetPasswordHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	if isURLFromAdminAPI(r) {
+		h.serveHTTPNonAuthflow(w, r)
+	} else {
+		h.serveHTTPAuthflow(w, r)
+	}
+}
+
+func (h *AuthflowV2ResetPasswordHandler) serveHTTPNonAuthflow(w http.ResponseWriter, r *http.Request) {
+	code := r.URL.Query().Get("code")
+	makeHTTPHandler := func(handler func(w http.ResponseWriter, r *http.Request) error) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			var err error
+			if code == "" {
+				err = newCodeRequiredErr()
+			} else {
+				err = handler(w, r)
+			}
+			if err != nil {
+				if apierrors.IsAPIError(err) {
+					// Still uses AuthflowController to render error because of same logic
+					h.Controller.ErrorRenderer.RenderError(w, r, err)
```

Comment:
Inject `ErrorRenderer` to the handle directly and use it, instead of access it through authflow controller.

Reply:
I see. Seems no need if we use `webapp controller` just like `pkg/auth/handler/webapp/authflowv2/select_account.go`